Add updated NEST examples to ReadtheDocs

jessica-mitchell opened the pull request:
Previously, the examples  were linked to the old website and some scripts needed to be updated to work with newest versions of NEST. This is related to issue #1271.

Here the updated examples were generated via sphinx-gallery (which produces an rst, py and ipynb for each example) and are moved to the doc/examples folder.

**This is an interim solution to have the updated examples available for users** 

Future PRs are planned to automate/test the examples, include the output images in the gallery of examples, and improve the organization of the examples - as it is now, they are all listed together, with related examples next to each other.

Tagging @hakonsbm @heplesser since you were involved in discussions regarding issue #1271 

See output here: https://nest-test.readthedocs.io/en/pyexamples_update/examples/index.html

heplesser commented on doc/examples/index.rst:
```diff
@@ -1,5 +1,136 @@
+.. code-block:: python
+
+   mpirun -n num_mpi_prc
+   python example.py
```
This needs to be on one line, otherwise `mpirun` will fail complaining about not having been given an executable.

heplesser commented on doc/examples/index.rst:
```diff
@@ -1,5 +1,136 @@
+input. This can be defined in the file ``stimulus_params.py``.
+
+Tested configuration: This version has been tested with NEST 2.10.0,
+Python 2.7.12 and 3.5.2, NumPy 1.11.2
```
We keep telling people to use current versions of NEST. Could you test if this works with 2.18? I assume all our examples are auto-tested before a release, so I hope this just needs an updated version number (or two) here.

jessica-mitchell replied:
@heplesser The microciricuit example.py ran for me with 2.18. This text was leftover I guess from when this example was originally made. I'd rather avoid writing 'tested versions', and prefer just remove this completely. (Unless we find an automated way of showing this).

As for the formatting errors, with the gap_junctions example- what exactly do you mean? If I need to regenerate the output files, it could take a bit of time

heplesser commented:
@jessica-mitchell The formatting errors are "blue boxes" that shouldn't be there, e.g. the "1" and "Authors" in the [gap junction example](https://nest-test.readthedocs.io/en/pyexamples_update/examples/gap_junctions_inhibitory_network.html) and multiple lines of normal text offset in boxes in the [multimeter to file example](https://nest-test.readthedocs.io/en/pyexamples_update/examples/multimeter_file.html), see also screenshots below.

![Screenshot 2019-09-04 at 22 58 33](https://user-images.githubusercontent.com/1059644/64291222-f164b180-cf67-11e9-9fc5-44228e31df9c.png)

![Screenshot 2019-09-04 at 22 58 51](https://user-images.githubusercontent.com/1059644/64291238-f7f32900-cf67-11e9-96b9-04713de7f1dd.png)

jessica-mitchell commented:
@heplesser The 1 for the reference is correct, it's how references are formatted in restructured text in this theme.  
The see also and authors sections were sections we wanted to include in the examples, but these are not completed yet. I can remove the empty ones  for the time being if that's preferred.

The other sections are mistakes in formatting the python scripts, which I can go through and fix.

heplesser commented:
@jessica-mitchell Thank you for the explanations! I think it makes sense to remove empty sections to reduce user confusion. Once the sections are filled, shouldn't "See also" and "Authors" be styled the same way, i.e., bold-face large?

Is the formatting of the reference numbers a choice or technical necessity? The current style looks so different from normal reference formatting that I find it rather confusing. I think a plain "[1]" in front of the citation would work better.

jessica-mitchell commented:
@heplesser I have now modified the files to fix up some formatting issues. I also changed the CSS a bit to make the reference number appear a bit nicer, but they're not in brackets. The authors, if provided, have a subsection similar to see also.

heplesser commented:
@jessica-mitchell Looks much better now, but in some random checks I noticed a few more problems:

- The reference is incomplete in https://nest-test.readthedocs.io/en/pyexamples_update/examples/brette_gerstner_fig_3d.html
- The 2nd reference looks funny ("2(1,2)") in https://nest-test.readthedocs.io/en/pyexamples_update/examples/precise_spiking.html

Could you check all examples once more to make sure the references are proper?

jessica-mitchell commented:
@heplesser 
I've updated the references but the reason for this syntax 1(1,2) actually is a back reference to every instance the reference is mentioned in the file.

heplesser commented:
I wonder how many readers will understand the 1(1,2) syntax before accidentally clicking on the links ;).
